src/modules/user/user.usecase.spec.ts

```diff
@@ -249,6 +266,105 @@ describe('User use cases', () => {
     });
   });
 
+  describe('Unblocking user account', () => {
+    describe('Request Account unblock', () => {
+      const fixedSystemCurrentDate = new Date('2020-02-19');
+
+      beforeAll(async () => {
+        jest.useFakeTimers();
+        jest.setSystemTime(fixedSystemCurrentDate);
+      });
+
+      afterAll(async () => {
+        jest.useRealTimers();
+      });
+
+      it('When user does not exist, then fail', async () => {
+        const userFindByEmailSpy = jest.spyOn(userRepository, 'findByEmail');
+        const email = 'email@test.com';
+        userFindByEmailSpy.mockReturnValueOnce(null);
+
+        await expect(
+          userUseCases.sendAccountUnblockEmail(email),
+        ).rejects.toThrow(NotFoundException);
+      });
+
+      it('When user user exists, then user lastPasswordChangedAt is updated', async () => {
+        const userFindByEmailSpy = jest.spyOn(userRepository, 'findByEmail');
+        const userUpdateSpy = jest.spyOn(userRepository, 'updateByUuid');
+        const configServiceGetSpy = jest.spyOn(configService, 'get');
+        const email = 'email@test.com';
+        userFindByEmailSpy.mockResolvedValueOnce(user);
+        configServiceGetSpy.mockReturnValue('secret');
+
+        await userUseCases.sendAccountUnblockEmail(email);
+
+        expect(SignWithCustomDuration).toHaveBeenCalledWith(
+          {
+            payload: {
+              uuid: user.uuid,
+              email: user.email,
+              action: AccountTokenAction.Unblock,
+            },
+            iat: getTokenDefaultIat(),
+          },
+          'secret',
+          '48h',
+        );
+        expect(userUpdateSpy).toHaveBeenCalledWith(user.uuid, {
+          lastPasswordChangedAt: fixedSystemCurrentDate,
+        });
+      });
+    });
+
+    describe('Unblock account', () => {
+      it('When user does not exist, then fail', async () => {
+        const userFindByUuidSpy = jest.spyOn(userRepository, 'findByUuid');
+        userFindByUuidSpy.mockReturnValueOnce(null);
+
+        await expect(userUseCases.unblockAccount(user.uuid)).rejects.toThrow(
+          BadRequestException,
+        );
+      });
+
+      it('When token is older than lastPasswordChangedAt, then fail', async () => {
```

Comment:
Is not older, is previous to the date, I find this legend misleading as 'older' often means after when talking about time